pkg/deployments/tasks/2022XXXX-tetu-rebalanced-linear-pool/test/test.fork.ts

```diff
@@ -0,0 +1,307 @@
+  const FINAL_LOWER_TARGET = fp(0.2e7);
+  const FINAL_UPPER_TARGET = fp(5e7);
+
+  let pool: Contract;
+  let poolId: string;
+
+  before('run task', async () => {
+    task = new Task('2022XXXX-tetu-rebalanced-linear-pool', TaskMode.TEST, getForkedNetwork(hre));
+    await task.run({ force: true });
+    factory = await task.deployedInstance('TetuLinearPoolFactory');
+  });
+
+  before('load signers', async () => {
+    [, owner, other] = await getSigners();
+    holder = await impersonate(USDT_HOLDER, fp(100));
+  });
+
+  before('setup contracts', async () => {
+    vault = await new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre)).deployedInstance('Vault');
+    usdt = await task.instanceAt('IERC20', USDT);
+    await usdt.connect(holder).approve(vault.address, MAX_UINT256);
+  });
+
+  enum LinearPoolState {
+    BALANCED,
+    MAIN_EXCESS,
+    MAIN_LACK,
+  }
+
+  function itRebalancesThePool(expectedState: LinearPoolState) {
+    it('rebalance the pool', async () => {
+      const { lowerTarget, upperTarget } = await pool.getTargets();
+
+      const { cash } = await vault.getPoolTokenInfo(poolId, USDT);
+      const scaledCash = cash.mul(USDT_SCALING);
+
+      let fees;
+      if (scaledCash.gt(upperTarget)) {
+        expect(expectedState).to.equal(LinearPoolState.MAIN_EXCESS);
+
+        const excess = scaledCash.sub(upperTarget);
+        fees = excess.mul(SWAP_FEE_PERCENTAGE).div(FP_ONE);
+      } else if (scaledCash.lt(lowerTarget)) {
+        expect(expectedState).to.equal(LinearPoolState.MAIN_LACK);
+
+        const lack = lowerTarget.sub(scaledCash);
+        fees = lack.mul(SWAP_FEE_PERCENTAGE).div(FP_ONE);
+      } else {
+        expect(expectedState).to.equal(LinearPoolState.BALANCED);
+
+        fees = 0;
+      }
+
+      const initialRecipientMainBalance = await usdt.balanceOf(other.address);
+
+      const governance = await impersonate(TETU_GOVERNANCE);
+      const controller = await task.instanceAt('ITetuController', TEUT_CONTROLLER);
+      await controller.connect(governance).changeWhiteListStatus([rebalancer.address], true);
+
+      if (expectedState != LinearPoolState.BALANCED) {
+        await rebalancer.connect(holder).rebalance(other.address);
+      } else {
+        await rebalancer.connect(holder).rebalanceWithExtraMain(other.address, 5);
+      }
+      const finalRecipientMainBalance = await usdt.balanceOf(other.address);
+
+      if (fees > 0) {
+        // The recipient of the rebalance call should get the fees that were collected (though there's some rounding
+        // error in the main-wrapped conversion).
+        expect(finalRecipientMainBalance.sub(initialRecipientMainBalance)).to.be.almostEqual(
+          fees.div(USDT_SCALING),
+          // because of rounding at TetuVault we can't provide better precision than 0.001
+          // But it should be fine because of rebalace will be done by Tetu team and fees at Polygon are low.
+          0.001
```

Comment:
This precision was decreased compared to AAVE version. I think it is ok but want you take a look.